lib/_tls_common.js

```diff
@@ -153,6 +153,38 @@ exports.createSecureContext = function createSecureContext(options) {
     }
   }
 
+  const engineKey = options.engineKey;
+  const privateKeyEngine = options.privateKeyEngine;
+  if (engineKey && privateKeyEngine) {
+    if ((typeof options.engineKey === 'string') &&
```

Comment:
Don't extract the property from the `options` object twice: Use `engineKey` here, not `options.engineKey`. (In JavaScript, there is no guarantee that `options.engineKey === options.engineKey`.)